### PR TITLE
Fibonacci sequence

### PR DESCRIPTION
**Pull request by avoloshchuk567**
Program of Fibonacci sequence definition is created.

### CODE REVIEW

File: src/main/java/fibonacciSequence/FibonacciSequenceDefinition.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    private List<Integer> fibonacciSequence = new ArrayList<>();
+
+    public List<Integer> defineFibonacciSequence(int number) {
+        this.number = number;
+        int sum;
+
+        if (number == 1) {
+            fibonacciSequence.add(1);
+            return fibonacciSequence;
+        } else if (number >= 2) {
```
Comment (uvidit) [inline src/main/java/fibonacciSequence/FibonacciSequenceDefinition.java]: let's try to optimize it somehow

File: src/main/java/fibonacciSequence/FibonacciSequenceDefinition.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            fibonacciSequence.add(1);
+            fibonacciSequence.add(1);
+
+            for (int i = 2; i < number; i++) {
+                sum = fibonacciSequence.get(i - 1) + fibonacciSequence.get(i - 2);
+                fibonacciSequence.add(sum);
```
Comment (uvidit) [inline src/main/java/fibonacciSequence/FibonacciSequenceDefinition.java]: could be simplified

File: src/test/java/fibonacciSequenceTests/FibonacciSequenceTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                {10, new Integer[]{1, 1, 2, 3, 5, 8, 13, 21, 34, 55}},
+                {1, new Integer[]{1}},
+                {2, new Integer[]{1, 1}},
+        };
+    }
+
+    @Test
+    @UseDataProvider("data")
+    public void sequenceOfTen(int number, Integer[] list) {
+        assertThat(definition.defineFibonacciSequence(number), Matchers.contains(list));
```
Comment (uvidit) [inline src/test/java/fibonacciSequenceTests/FibonacciSequenceTest.java]: dataprovider items and input parameters could be simplified to getting the list size

**Review by uvidit — CHANGES_REQUESTED** [review]

File: src/main/java/fibonacciSequence/FibonacciSequenceDefinition.java (hunk 1 above)
Comment (avoloshchuk567) [reply src/main/java/fibonacciSequence/FibonacciSequenceDefinition.java]: Optimized in b7f7b7eb3c7316a5b476c68f07f63ae64ca984d3. Changed to Stream.iterate()...

File: src/main/java/fibonacciSequence/FibonacciSequenceDefinition.java (hunk 2 above)
Comment (avoloshchuk567) [reply src/main/java/fibonacciSequence/FibonacciSequenceDefinition.java]: Optimized in b7f7b7eb3c7316a5b476c68f07f63ae64ca984d3. 
Changed to Stream.iterate()...

File: src/test/java/fibonacciSequenceTests/FibonacciSequenceTest.java (hunk 3 above)
Comment (avoloshchuk567) [reply src/test/java/fibonacciSequenceTests/FibonacciSequenceTest.java]: Changed in b7f7b7eb3c7316a5b476c68f07f63ae64ca984d3.
Redundant input 'int number' parameter was removed from method signature, and changed to list.length in assertion.

**Review by uvidit — APPROVED** [review]

**Pull request merged**